src/test/java/com/digirati/taxonomy/manager/lookup/LookupTest.java

```diff
@@ -0,0 +1,117 @@
+                matches,
+                match("row", 2, 4, fight, line),
+                match("sake", 24, 27, purpose, riceWine),
+                match("row", 62, 64, fight, line),
+                match("sake", 69, 72, purpose, riceWine));
+    }
+
+    @Test
+    void searchShouldBeAbleToHandleMatchedTermsWithNoAttachedConcepts() {
+        // Given
+        Lookup underTest = new Lookup("row", "sake");
+
+        // When
+        Collection<Match> matches =
+                underTest.search(
+                        "a row broke out for the sake of working out who could drink a row of sake the fastest");
+
+        // Then
+        assertMatches(
+                matches,
+                match("row", 2, 4),
+                match("sake", 24, 27),
+                match("row", 62, 64),
+                match("sake", 69, 72));
+    }
+
+    @Test
+    void rebuildAutomatonShouldReplaceOriginalTermSet() {
+        // Given
+        Lookup underTest = new Lookup("row", "sake");
+        underTest.rebuildAutomaton("drink", "fastest");
+
+        // When
+        Collection<Match> matches =
+                underTest.search(
+                        "a row broke out for the sake of working out who could drink a row of sake the fastest");
+
+        // Then
+        assertMatches(matches, match("drink", 54, 58), match("fastest", 78, 84));
+    }
+
+    @Test
+    void clearConceptLookupTableShouldRemoveAnyExistingConceptMappings() {
+        // Given
+        Lookup underTest = new Lookup("row", "sake");
+        ConceptLookupTable conceptLookupTable = underTest.getConceptLookupTable();
+        conceptLookupTable.put(
+                "row",
+                new Concept("fight", Lists.newArrayList("row")),
+                new Concept("line", Lists.newArrayList("row")));
+        conceptLookupTable.put(
+                "sake",
+                new Concept("purpose", Lists.newArrayList("sake")),
+                new Concept("rice wine", Lists.newArrayList("sake")));
+
+        underTest.clearConceptLookupTable();
+
+        // When
+        Collection<Match> matches =
+                underTest.search(
+                        "a row broke out for the sake of working out who could drink a row of sake the fastest");
+
+        // Then
+        assertMatches(
+                matches,
+                match("row", 2, 4),
+                match("sake", 24, 27),
+                match("row", 62, 64),
+                match("sake", 69, 72));
+    }
+
+    private void assertMatches(Collection<Match> actual, Match... expected) {
+        MatcherAssert.assertThat(actual, Matchers.contains(expected));
+    }
+
+    private Match match(String term, int startIndex, int endIndex) {
+        return new Match(new Emit(startIndex, endIndex, term), null);
+    }
+
+    private Match match(String term, int startIndex, int endIndex, Concept... concepts) {
+        return new Match(new Emit(startIndex, endIndex, term), Lists.newArrayList(concepts));
```

Comment:
The reason for this is the same as the reason for the null above - when no matching concepts are found in the concept lookup table, the Match has a null for the list of concepts but this puts an empty list which, with the production code as it currently is, will cause this `Match` to not be equal to the actual one. I had avoided changing the production code just to cater for the tests, but because this is new code I'm not too precious about that point

Reply:
So the tests only pass here because production code allows you to use the API incorrectly. The `Match` constructor should be changed to reject `null` as a value.